src/dropdown.ts

```diff
@@ -587,30 +591,18 @@ export default class Dropdown extends LitElement implements FormControl {
 
   formResetCallback(): void {
     for (const option of this.#optionElements) {
-      const isInitiallySelected = option.hasAttribute('selected');
+      // `#onOptionsSelectedChange()` is called when an option is selected. It updates
+      // `this.selectedAndEnabledOptions`. Deselecting every option before reselecting
+      // those in `value` ensures tags appear in the same order they were initially.
+      option.selected = false;
 
-      if (!isInitiallySelected) {
-        option.selected = false;
+      const isInitiallySelected =
+        option.hasAttribute('selected') && !option.disabled;
+
+      if (isInitiallySelected) {
+        option.selected = true;
       }
     }
-
-    const initiallySelectedOptionElementsWithValue =
-      this.#optionElements.filter((option) => {
-        return option.hasAttribute('selected');
-      });
-
-    // Even with a single-select, there's nothing to stop developers from adding
-    // a `selected` attribute to more than one option. How native handles it
-    // is to choose the last selected option. This mimics that behavior, which
-    // seems reasonable.
-    const lastValue = initiallySelectedOptionElementsWithValue.at(-1)?.value;
-
-    this.#value =
```

Comment:
No need to set `#value` here. `#onOptionsSelectedChange()` can do it and is now the source of truth for setting `this.#value` when an option is selected or deselected.